card/lib/card/format/content.rb

```diff
@@ -1,15 +1,21 @@
 class Card
   class Format
     module Content
-      def process_content override_content=nil, content_opts=nil
+      def process_content override_content=nil, content_opts=nil, &block
         content = override_content || render_raw || ""
         content_object = get_content_object content, content_opts
-        content_object.process_each_chunk do |chunk_opts|
-          content_nest chunk_opts
-        end
+        content_object.process_chunks(&block)
         content_object.to_s
       end
 
+      def safe_process_content override_content=nil, content_opts=nil, &block
```

Comment:
This should probably have a short comment to explain "safe" to our later selves.